### PR TITLE
Invisible Plugin in uxp

### PR DESCRIPTION
**Pull request by ThisIsMrAli**
I have worked on this [issue](https://github.com/AdobeDocs/uxp-photoshop-plugin-samples/issues/47) and I could manage to migrate the Invisible CEP extension to UXP. Please take a look and let me know if there is any problem.
@pklaschka @ErinFinnegan

### CODE REVIEW

**Comment by ErinFinnegan** [comment]
🎉🎉 Thank you!! I can’t wait to look! (I’m on vacation this week.)

**Comment by ThisIsMrAli** [comment]
> 🎉🎉 Thank you!! I can’t wait to look! (I’m on vacation this week.)

Great Thanks! Looking forward to it!

File: invisible-plugin-sample/index.js
Comment (pklaschka) [inline invisible-plugin-sample/index.js:12]:
```suggestion
    },
});
```

File: invisible-plugin-sample/README.md
Comment (pklaschka) [inline invisible-plugin-sample/README.md:3]: Could you please elaborate a little bit about what an invisible plugin is here? This doesn't have to be a wall of text, but just describe in one or two sentences what the plugin does.

**Review by pklaschka — CHANGES_REQUESTED** [review]
Great pull request! And sorry for getting back to you so late on this.

I only have a couple of minor requests, and after that, we should be good to merge this one. Please let me know if you have any concerns about any of my comments. Thank you for your contribution!

File: invisible-plugin-sample/README.md
Comment (ThisIsMrAli) [reply invisible-plugin-sample/README.md:3]: @pklaschka thanks for your review. is it possible for you to update the description if i explain it here?

**Review by gregbenz — COMMENTED** [review]
core.showAlert is an async function, shouldn't this be created as async create() and then await the alert?

**Comment by ThisIsMrAli** [comment]
@gregbenz Since its just a sample that shows the invisible plugin has loaded, i dont think so that its necessary to use await.

**Comment by ThisIsMrAli** [comment]
@pklaschka i've fixed the mentioned problems. Please take a look and let me know.

**Comment by gregbenz** [comment]
What triggers the the JS to get to plugin.create()? It seems like panels are normally not invoked prior to showing them during a PS session. And when I install a packaged CCX of this example, the alert does not show after PS startup, so it seems it never runs (other than when installed or enabled by CCD). As far as I can see, this plugin ultimately results in just showing a greyed menu item under Plugins once installed under the PS v24.

**Comment by gregbenz** [comment]
> @gregbenz Since its just a sample that shows the invisible plugin has loaded, i dont think so that its necessary to use await.

Agreed, not critical, but someone might miss that the create method is async.

File: invisible-plugin-sample/README.md
Comment (amandahuarng) [inline invisible-plugin-sample/README.md:3]: Swap for this - "This plugin is an example of an invisible UXP plugin. An invisible plugin remains hidden during its whole life cycle -- meaning it does not have any UI or (persistent) menu options."

**Review by amandahuarng — APPROVED** [review]
looks good to me as well. I left one small comment in the description in the README .

**Comment by amandahuarng** [comment]
Agree with @gregbenz comment on async/await for the alert. If it is not consistently showing up in your UI, let's call that out in the README explicitly so people understand the expected behavior of the sample. @ThisIsMrAli

File: invisible-plugin-sample/README.md
Comment (ThisIsMrAli) [reply invisible-plugin-sample/README.md]: @amandahuarng Done

**Comment by ThisIsMrAli** [comment]
> CCD)

I checked it with UXP devtool and it works well, I will check the CCX package too. Thanks for letting me now.

**Comment by ThisIsMrAli** [comment]
as @gregbenz mentioned,  plugin.create() triggers only the first time that you opened PS after you installed the plugin. I need to find another solution for this. i appreciate any help for this. @gregbenz @amandahuarng

**Comment by amandahuarng** [comment]
@ThisIsMrAli Pablo and I tested in on our machines and found that for Photoshop - after **first installation** (whether locally or via CCX), the alert will show up. Please update your manifest to use v5 (and min version field) as shown in the screenshot below. 

That should fix the alert not popping up when installing via CCX. We can merge this PR if we update the README to make it clear the `entrypoints.plugin.create` is called only once upon plugin installation. Therefore, if you want to see the alert again, you'll have to uninstall and reinstall in UDT (or disable then enable again on CCD).


Screen Shot 2022-11-22 at 10 33 45 AM

**Comment by ThisIsMrAli** [comment]
@amandahuarng Thanks Amanda for your help! I have update the manifest to version 5. Since you got a better explanation about the Plugin.create function, would you please update the Readme.md file and merge the request?

**Comment by amandahuarng** [comment]
> @amandahuarng Thanks Amanda for your help! I have updated the manifest to version 5. Since you have got a better explanation about the Plugin.create function, would you please update the Readme.md file and merge the request?

Yes, I've added notes to the README and am merging this sample.

**Pull request merged**
